crates/adapters/src/client_cache.rs

```diff
@@ -0,0 +1,386 @@
+//! HTTP client cache for optimized connection management
+//!
+//! Provides per-solver client instances with connection pooling and keep-alive optimization.
+
+use dashmap::DashMap;
```

Comment:
Client cache optimisations to keep alive connects between requests.
Used within adapters